mlrun/api/crud/runs.py

```diff
@@ -41,6 +42,11 @@ def store_run(
         project: str = mlrun.mlconf.default_project,
     ):
         project = project or mlrun.mlconf.default_project
+
+        # Some runtimes do not use the submit job flow, so their notifications are not masked.
+        # Redact notification params if not concealed with a secret
+        mlrun.api.api.utils.mask_notification_params_on_task(data, "redact")
```

Comment:
Are we guaranteed that for every other runtimes, the `store` will only happen after the `submit`? I'm concerned there might be a flow (maybe for specific runtime types) where the run is generated, saved, and then submitted. My thinking was to redact the params only when storing runtimes of local type (`local`, `handler`). WDYT?

Reply:
redacting for specific runtimes is problematic in my eyes because it's not only local and handler but also dask and it will require maintaining the list of runtimes this applies for which I prefer to avoid.
I think either way these values should not be stored in the DB and it should apply for all runtimes. If some runtime uses the run object from the DB than we have a different bug which needs to be addressed differently.